src/subsai/models/whisper_api_model.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Whisper API Model
+
+See [openai/whisper](https://platform.openai.com/docs/guides/speech-to-text)
+"""
+
+import os
+import ffmpeg
+from subsai.models.abstract_model import AbstractModel
+from subsai.utils import _load_config
+from openai import OpenAI
+from pysubs2 import SSAFile
+
+def convert_video_to_audio_ffmpeg(video_file, output_ext="mp3"):
+    # Construct the output file name
+    filename, ext = os.path.splitext(video_file)
+    output_file = f"{filename}.{output_ext}"
+    
+    # Execute the ffmpeg conversion
+    (
+        ffmpeg
+        .input(video_file)
+        .output(output_file)
+        .overwrite_output()
+        .run(quiet=True)
+    )
+
+    return output_file
+
+
+class WhisperAPIModel(AbstractModel):
+    model_name = 'openai/whisper'
+    config_schema = {
+            # load model config
+            'model_type': {
+                'type': list,
+                'description': "One of the official model names listed by `whisper.available_models()`, or "
+                               "path to a model checkpoint containing the model dimensions and the model "
+                               "state_dict.",
+                'options': ['whisper-1'],
+                'default': 'whisper-1'
+            },
+            'api_key': {
+                'type': str,
+                'description': "Your OpenAI API key",
+                'options': None,
+                'default': None
+            },
+        }
+
+    def __init__(self, model_config):
+        # config
+        self.model_type = _load_config('model_type', model_config, self.config_schema)
+        self.api_key = _load_config('api_key', model_config, self.config_schema)
+
+        self.client = OpenAI(api_key=self.api_key)
+        
+
+    def transcribe(self, media_file) -> str:
+        audio_file_path = convert_video_to_audio_ffmpeg(media_file)
```

Comment:
Do you think this conversion is needed ? 
Their docs said: 
> the following input file types are supported: mp3, mp4, mpeg, mpga, m4a, wav, and webm.

Have you tried a video file without conversion and didn't work ?

Reply:
No I didn't but I assumed due to the 25mb upload limit that this wouldn't be a useful use case

Reply:
Yeah, but I was talking about this function `convert_video_to_audio_ffmpeg`, from their docs I can see that they support all video and audio format, so you don't really need to convert the video to audio using ffmpeg I guess! as long as the media file does not exceed 25MB you can send it! 
If you can run the test with a file, for example the one in `assets/video/test1.webm` and it worked without conversion then we can get rid of this function and optimize the code more.

Reply:
Why would we remove it? If I want to transcribe a 1GB+ video file I'd rather not send the video over the internet in 25MB chunks

I can remove it for files which are below 25MB if that's what you mean

Reply:
True! extracting the audio from the video will make the file smaller. which is good in this case.
It's okey just leave it, even for files below 25MB, transcoding them to mp3 may compress them a bit anyways.